modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -18,18 +18,76 @@ import * as Parents from '../dom/Parents';
 import * as TableCellSelection from '../selection/TableCellSelection';
 import * as DeleteElement from './DeleteElement';
 import * as TableDeleteAction from './TableDeleteAction';
+import * as TableDeleteUtils from './TableDeleteUtils';
 
 const freefallRtl = (root: SugarElement<Node>): Optional<SugarElement<Node>> => {
   const child = SugarNode.isComment(root) ? Traverse.prevSibling(root) : Traverse.lastChild(root);
   return child.bind(freefallRtl).orThunk(() => Optional.some(root));
 };
 
-const emptyCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[]) => {
+const shallowCopyRange = (rng: Range) => ({
+  startContainer: rng.startContainer,
+  startOffset: rng.startOffset,
+  endContainer: rng.endContainer,
+  endOffset: rng.endOffset
+});
+
+const removeContentFromCells = (cells: SugarElement<HTMLTableCellElement>[]) =>
   Arr.each(cells, PaddingBr.fillWithPaddingBr);
-  editor.selection.setCursorLocation(cells[0].dom, 0);
+
+/*
+ * Runs when
+ * - the start and end of the selection is contained within the same table (called directly from deleteRange)
+ * - part of a table and content outside is selected
+ */
+const emptySingleTableCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], outsideRng: Optional<Range>) => {
+  const selection = editor.selection;
+  const rng = selection.getRng();
+  const { startContainer, startOffset, endContainer } = shallowCopyRange(rng);
+  const { startInTable, partialSelection } = TableDeleteUtils.getTableDetailsFromRange(rng, SugarElement.fromDom(editor.getBody()));
+  const outsideBlock = Optionals.someIf(partialSelection, SugarElement.fromDom(editor.dom.getParent(startInTable ? endContainer : startContainer, editor.dom.isBlock)));
+
+  // Remove content from selected cells
+  removeContentFromCells(cells);
+  // Delete all content outside of the table that is in the selection
+  outsideRng.each((rng) => rng.deleteContents());
+
+  // Handle block outside the table if it is empty since rng.deleteContents leaves it
+  // Set the appropriate cursor location
+  outsideBlock
+    .filter(Empty.isEmpty)
+    .fold(() => {
+      if (startInTable) {
+        selection.setCursorLocation(cells[0].dom, 0);
+      } else {
+        selection.setCursorLocation(startContainer, startOffset);
+      }
+    }, (emptyBlock) => {
+      if (startInTable) {
+        Remove.remove(emptyBlock);
+        selection.setCursorLocation(cells[0].dom, 0);
+      } else {
+        PaddingBr.fillWithPaddingBr(emptyBlock);
+        editor.selection.setCursorLocation(emptyBlock.dom, 0);
+      }
+    });
```

Comment:
```suggestion
  if (startInTable) {
    outsideBlock.each(Remove.remove);
    selection.setCursorLocation(cells[0].dom, 0);
  } else {
    outsideBlock.fold(
      () => selection.setCursorLocation(startContainer, startOffset),
      (emptyBlock) => {
        PaddingBr.fillWithPaddingBr(emptyBlock);
        selection.setCursorLocation(emptyBlock.dom, 0);
      }
    );
  }
```
This is a little cleaner and avoids some of the duplication. Note that this relies on moving the `filter` above as per my other suggestion.